### ISSUE TITLE
Profile.num_sound_downloads and Profile.num_pack_downloads not updating

### ISSUE THREAD

**Issue by ffont** [MTG/freesound#1156]
We recently added two new fields in the `Profile` model, namely `num_sound_downloads` and `num_pack_downloads`. Originally, these fields were updated using Django signals after download events (see  [this code](https://github.com/MTG/freesound/commit/938a669904dfddbeb5d181370972f8fa8e27def1), changes in `sounds/models.py` signals). However, after a bad merge (conflicts not properly fixed), the lines of code which update `Profile.num_sound_downloads` and `Profile.num_pack_downloads` were removed. Because we have no test to check `Profile.num_sound_downloads` and `Profile.num_pack_downloads` are ok and we don't (yet) make an extensive use of these fields, we did not realize until now about the bad merge.

We should re-implement functionality (including tests!) so that `Profile.num_sound_downloads` and `Profile.num_pack_downloads` are updated when users download sounds and packs. This should happen for sound downloads ([here](https://github.com/MTG/freesound/blob/4786f2b62f19ad680609d6554ca9ab192830f7cd/sounds/models.py#L1064-L1076)) and also for pack downloads ([here](https://github.com/MTG/freesound/blob/4786f2b62f19ad680609d6554ca9ab192830f7cd/sounds/models.py#L1091-L1101)).

Once this is done, `Profile.num_sound_downloads` and `Profile.num_pack_downloads` fields will be reliable again, and we can revert that [temporary change with the "show about"](https://github.com/MTG/freesound/pull/1155/commits/06c3fa9adb9532068d9ebfcc16fa8027c0cff9ff) condition.

Also, after this is done an optimization could be implemented in the [`downloaded_sounds`](https://github.com/MTG/freesound/blob/master/accounts/views.py#L703) and the [`downloaded_packs`](https://github.com/MTG/freesound/blob/master/accounts/views.py#L718) views as `user.profile.num_sound_downloads` and `user.profile.num_pack_downloads` can be passed to the paginator to avoid one extra query to the DB to get the total count of objects (similarly as is done [here](https://github.com/MTG/freesound/blob/master/sounds/views.py#L924)).

Finally, to prevent `Profile.num_sound_downloads` and `Profile.num_pack_downloads` of  going out of sync for unexpected reasons, we have a [management command](https://github.com/MTG/freesound/blob/master/general/management/commands/report_count_statuses.py) which makes sure all fields storing counts of objects are in sync (and fixes them if they are not). However, this command is **not currently** taking care of `Profile.num_sound_downloads` and `Profile.num_pack_downloads` and therefore it should be updated. This is explained in another issue here: https://github.com/MTG/freesound/issues/1113.
